### PR TITLE
Handle manufacturer data length for diagnosis in USI format

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,30 +71,26 @@ set(PNET_MAX_LOG_BOOK_ENTRIES   16
   CACHE STRING "")
 set(PNET_MAX_ALARMS             3
   CACHE STRING "Per AR and queue. One queue for high and one for low priority alarms.")
+set(PNET_MAX_ALARM_PAYLOAD_DATA_SIZE 28
+  CACHE STRING "Min 24 or PNET_MAX_DIAG_MANUF_DATA_SIZE + approx 12. Max is 1408")
```
Comment (olbjo) [inline CMakeLists.txt:75]: Maybe we can remove use of "approx"?

File: CMakeLists.txt [hunk 2]
```diff
@@ -71,30 +71,26 @@ set(PNET_MAX_LOG_BOOK_ENTRIES   16
...
 set(PNET_MAX_DIAG_ITEMS         200
   CACHE STRING "Total, per device. Max is 65534 items")
-
+set(PNET_MAX_DIAG_MANUF_DATA_SIZE 16
+  CACHE STRING "Min approx 7 for tests. Max is 1396")
```
Comment (olbjo) [inline CMakeLists.txt:79]: Remove "approx"?

File: test/test_diag.cpp [hunk 3]
```diff
@@ -593,6 +595,7 @@ TEST_F (DiagTest, DiagRunTest)
       TEST_SLOT_IDENT,
       TEST_SUBSLOT_IDENT,
       TEST_DIAG_USI_CUSTOM,
+      7,
       (uint8_t *)"Bjarn2");
```
Comment (olbjo) [inline test/test_diag.cpp:599]: Replace "Bjarn" with something anonymous. Search for and replace "Bjarn"

**Review by olbjo — APPROVED** [review]
Great!
Some minor comments only

File: CMakeLists.txt (hunk 1 above)
Comment (pyhys) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (pyhys) [reply CMakeLists.txt]: Done

File: test/test_diag.cpp (hunk 3 above)
Comment (pyhys) [reply test/test_diag.cpp]: Done

**Pull request merged**